rust/syntect-plugin/src/main.rs

```diff
@@ -184,6 +193,76 @@ impl<'a> PluginState<'a> {
         ctx.reset();
         ctx.schedule_idle(0);
     }
+
+    /// Checks if a newline has been inserted, and if so inserts whitespace
+    /// as necessary.
+    fn do_indentation(&mut self, ctx: &mut PluginCtx<State>, start: usize,
+                      end: usize, new_len: usize, rev: usize,
+                      text: &Option<String>) -> Option<Value> {
+        // don't touch indentation if this is not a simple edit
+        if end != start { return None }
+        let is_newline = {
+            let line_ending = &ctx.get_config().line_ending;
+                text.as_ref()
+                .map(|t| t.ends_with(line_ending))
+                .unwrap_or(false)
+        };
+
+        if is_newline {
+            let line_num = ctx.find_offset(start).err();
```

Comment:
I think this fails to indent if the edit is to insert, say, "{\n" at the beginning of a line. The value should come back in the ok part of the result. This is unlikely to happen (the actual case likely to trigger it is a paste, for which we need to plumb through disabling this).

Reply:
Good spot. I was mentally thinking of a "simple edit" as being one where start == end, but what we really care about is that this action only fires on the "insert_newline" RPC; so we can just do a string comparison on the text, and make this simpler.

Reply:
(I would leave the requirement that start == end; I'm not sure I want to touch indentation in the case where a selection is replaced with a newline.)